deployment/terraform/batch.tf

```diff
@@ -0,0 +1,75 @@
+resource "aws_batch_compute_environment" "default" {
+  depends_on = ["aws_iam_role_policy_attachment.batch_policy"]
+
+  compute_environment_name = "batch${var.environment}DefaultComputeEnvironment"
+  type                     = "MANAGED"
+  state                    = "ENABLED"
+  service_role             = "${aws_iam_role.container_instance_batch.arn}"
+
+  compute_resources {
+    type           = "SPOT"
+    bid_percentage = "${var.batch_default_ce_spot_fleet_bid_percentage}"
+    ec2_key_pair   = "${var.aws_key_name}"
+    image_id       = "${var.batch_ami_id}"
+
+    min_vcpus     = "${var.batch_default_ce_min_vcpus}"
+    desired_vcpus = "${var.batch_default_ce_desired_vcpus}"
+    max_vcpus     = "${var.batch_default_ce_max_vcpus}"
+
+    spot_iam_fleet_role = "${aws_iam_role.container_instance_spot_fleet.arn}"
+    instance_role       = "${aws_iam_instance_profile.container_instance.arn}"
+
+    instance_type = [
+      "${var.batch_default_ce_instance_types}",
+    ]
+
+    security_group_ids = [
+      "${aws_security_group.batch.id}",
+    ]
+
+    subnets = [
+      "${module.vpc.private_subnet_ids}",
+    ]
+
+    tags {
+      Name               = "BatchWorker"
+      ComputeEnvironment = "Default"
+      Project            = "${var.project}"
+      Environment        = "${var.environment}"
+    }
+  }
+}
+
+resource "aws_batch_job_queue" "default" {
+  name                 = "queue${var.environment}Default"
+  priority             = 1
+  state                = "ENABLED"
+  compute_environments = ["${aws_batch_compute_environment.default.arn}"]
+}
+
+data "template_file" "default_job_definition" {
+  template = "${file("job-definitions/default.json")}"
+
+  vars {
+    image_url      = "${module.ecr_repository.repository_url}:${var.image_tag}"
+    default_vcpus  = "${var.batch_default_jd_vcpus}"
```

Comment:
I think we should embed values for VCPUs and memory within the job definition json, as we do in Raster Foundry. 

I don't see what value we get out of abstracting it here, because the job definition will still need to be regenerated, and we're not going to want to reuse the same VCPU and memory configuration for each job definition.